### PR TITLE
Environment setup update

### PR DESCRIPTION
**Pull request by sergmetelin**
- Updating README.md to show a clean path to setup a simple node on clean install of Ubuntu 16.10.
- Added macOS build
- Added GMP for Ubuntu build
- Added account creation
- Added contract publish

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,16 +8,45 @@ Welcome to the EOS.IO source code repository!
...
+This project is written primarily in C++14 and uses CMake as its build system. An up-to-date C++ toolchain (such as Clang or GCC) and the latest version of CMake is recommended.
 
-### Installing Dependencies
-Eos has the following external dependencies, which must be installed on your system:
- - Boost 1.64
- - OpenSSL
- - LLVM 4.0 (Ubuntu users must install llvm-4.0 packages from https://apt.llvm.org/)
- - [secp256k1-zkp (Cryptonomex branch)](https://github.com/cryptonomex/secp256k1-zkp.git)
+Dependencies:
+* GCC 5.4.0 or Clang 4.0.0
```
Comment (elmato) [inline README.md]: I will leave Clang 4.0.0 alone, there has been some problems with GCC 5.4.0 even if the project successfully complies

Comment (sergmetelin) [reply README.md]: Ok, I'll update accordingly.

File: README.md [hunk 2]
```diff
@@ -26,6 +55,29 @@ make
 sudo make install
 ```
 
+And, lastly, LLVM 4.0:
+
+```
+cd ~
+wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
+apt-get install clang-4.0 lldb-4.0
```
Comment (thomasbcox) [inline README.md]: The line 
`apt-get install clang-4.0 lldb-4.0`
needs to read
`sudo apt-get install clang-4.0 lldb-4.0`

Comment (thomasbcox) [reply README.md]: Also, do we know minimum RAM, disk, and CPU for an acceptable node?

**Comment by thomasbcox** [comment]
This command
`WASM_LLVM_CONFIG=~/wasm-compiler/llvm/bin/llvm-config cmake -DBOOST_ROOT="$BOOST_ROOT" ..`
resulted in this error: 
`CMake Error at /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  Could NOT find GMP (missing: GMP_LIBRARIES GMP_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  libraries/fc/CMakeModules/FindGMP.cmake:50 (find_package_handle_standard_args)
  libraries/fc/CMakeLists.txt:20 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/tbcox/wasm-compiler/llvm/build/eos/build/CMakeFiles/CMakeOutput.log".`

**Comment by MatthewPappas** [comment]
You will need GMP installed. This is what I did, I know at least the 1st needs to be done, the other two I am not sure.

as root:  (feel free to modify these steps to match your other steps).
cd /tmp

wget ftp://gcc.gnu.org/pub/gcc/infrastructure/gmp-4.3.2.tar.bz2
tar xvjf gmp-4.3.2.tar.bz2
mkdir gmp-4.3.2/gmp-objs/
cd gmp-4.3.2/gmp-objs/
../configure --prefix=/usr/local --disable-shared
make install
cd ../..

followed by the package mpfr, as

wget ftp://gcc.gnu.org/pub/gcc/infrastructure/mpfr-2.4.2.tar.bz2
tar xvjf mpfr-2.4.2.tar.bz2
mkdir mpfr-2.4.2/mpfr-objs/
cd mpfr-2.4.2/mpfr-objs/
../configure --prefix=/usr/local --disable-shared --with-gmp=/usr/local
make install
cd ../..

and finally, the third package mpc, as

wget ftp://gcc.gnu.org/pub/gcc/infrastructure/mpc-0.8.1.tar.gz
tar xvzf mpc-0.8.1.tar.gz
mkdir mpc-0.8.1/mpc-objs/
cd mpc-0.8.1/mpc-objs/
../configure --prefix=/usr/local --disable-shared --with-gmp=/usr/local
make install
cd ../..

**Comment by bytemaster** [comment]
I'm not sure if I merged things properly with the docs, someone should review the README.

**Pull request merged**